builtin/providers/aws/waf_token_handlers.go

```diff
@@ -0,0 +1,49 @@
+package aws
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+type WafRetryer struct {
+	Connection *waf.WAF
+	Region     string
+}
+
+type withTokenFunc func(token *string) (interface{}, error)
+
+func (t *WafRetryer) RetryWithToken(f withTokenFunc) (interface{}, error) {
+	awsMutexKV.Lock(t.Region)
+
```

Comment:
Yes, indeed, thanks.

I had that there before, must have vanished while I was refactoring 🙈